### PR TITLE
docs:  prefetch tap fallback on slow connection and ignore slow connection option

### PR DESCRIPTION
**Pull request by smitbarmase**
Add docs for:
 
1. Fallback to `tap` prefetch strategy on slow connection, instead of bailing.
2. `ignoreSlowConnection` option to prefetch API, which lets manually prefetch even on data saver mode or slow connection.

 #### See astro PR [#9092](https://github.com/withastro/astro/pull/9092).

### CODE REVIEW

File: src/content/docs/en/guides/prefetch.mdx [hunk 1]
```diff
@@ -46,7 +46,7 @@ You can specify a strategy for an individual link by passing it to the `data-ast
 
 Each strategy is fine-tuned to only prefetch when needed and save your users' bandwidth. For example:
 
-- If a visitor is using [data saver mode](https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation/saveData) or has a [slow connection](https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation/effectiveType), prefetching will be turned off.
+- If a visitor is using [data saver mode](https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation/saveData) or has a [slow connection](https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation/effectiveType), prefetch will fallback to tap strategy.
```
Comment (bluwy) [inline src/content/docs/en/guides/prefetch.mdx:49]:
```suggestion
- If a visitor is using [data saver mode](https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation/saveData) or has a [slow connection](https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation/effectiveType), prefetch will fallback to the `tap` strategy.
```

File: src/content/docs/en/guides/prefetch.mdx [hunk 2]
```diff
@@ -116,6 +116,13 @@ prefetch('/about', { with: 'link' });
 
 The `prefetch()` API includes the same [data saver mode](https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation/saveData) and [slow connection](https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation/effectiveType) detection, so that it only prefetches when needed.
 
+To ignore slow connection detection, you can use `ignoreSlowConnection` option:
```
Comment (bluwy) [inline src/content/docs/en/guides/prefetch.mdx:119]:
```suggestion
To ignore slow connection detection, you can use the `ignoreSlowConnection` option:
```

**Review by bluwy — APPROVED** [review]
Minor wording nits, but it describes the feature from the core PR well 👍

**Review by sarah11918 — APPROVED** [review]
Thank you for this contribution, @smitbarmase ! The docs look good, and these will be added to the next minor release. Welcome to Team Docs!  :partying_face:

File: src/content/docs/en/guides/prefetch.mdx [hunk 3]
```diff
@@ -114,7 +114,14 @@ prefetch('/about', { with: 'fetch' });
 prefetch('/about', { with: 'link' });
 ```
 
-The `prefetch()` API includes the same [data saver mode](https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation/saveData) and [slow connection](https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation/effectiveType) detection, so that it only prefetches when needed.
+The `prefetch()` API includes the same [data saver mode](https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation/saveData) and [slow connection](https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation/effectiveType) detection, with fallback to the `tap` method.
```
[the remark this answers is not in the document]
Comment (smitbarmase) [reply src/content/docs/en/guides/prefetch.mdx:117]: Hi @sarah11918, to clarify, the `prefetch()` API is independent of tap/hover/viewport strategies as it is manually called by the user in a function. So, manually calling `prefetch()` can't fall back to tap, as it is not caused by user interaction like hover or viewport. I suggest we keep it as is for now, indicating that it bails on slow connections, and you can opt out of this behavior with the `ignoreSlowConnection` flag.

If "only prefetching when needed" seems vague, we can rephrase it to specify that it will bail on slow connections. Let me know what you think!

[the remark this answers is not in the document]
Comment (sarah11918) [reply src/content/docs/en/guides/prefetch.mdx:117]: Gotcha! In that case, then nope, I'm cool with the original! We can revert. Thanks for clarifying. Just wanted to make sure this wasn't overlooked.

Comment (sarah11918) [inline src/content/docs/en/guides/prefetch.mdx:117]:
```suggestion
The `prefetch()` API includes the same [data saver mode](https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation/saveData) and [slow connection](https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation/effectiveType) detection so that it only prefetches when needed.
```

[the remark this answers is not in the document]
Comment (bluwy) [reply src/content/docs/en/guides/prefetch.mdx]: Yup agree with Smit here!
